src/schemes/fluid/viscosity.jl

```diff
@@ -181,12 +169,15 @@ end
 end
 
 @inline function (viscosity::ViscosityMorris)(c, v_diff, pos_diff, distance, rho_mean,
-                                              rho_a, rho_b, h, grad_kernel)
-    (; epsilon, nu) = viscosity
+                                              rho_a, rho_b, h, grad_kernel, particle_system,
+                                              neighbor_system)
```

Comment:
I don't like that the viscosity, which is supposed to be an elementary function, needs systems. Can you instead pass `nu_a` and `nu_b`?

Reply:
Then we need to dispatch on the viscosity models in interaction.